testing/simapp/app.go

```diff
@@ -333,24 +339,31 @@ func NewSimApp(
 	transferModule := transfer.NewAppModule(app.TransferKeeper)
 	transferIBCModule := transfer.NewIBCModule(app.TransferKeeper)
 
+	// NOTE: the IBC mock keeper and application module is used only for testing core IBC. Do
+	// not replicate if you do not need to test core IBC or light clients.
+	mockModule := ibcmock.NewAppModule(scopedIBCMockKeeper, &app.IBCKeeper.PortKeeper)
+	mockIBCModule := ibcmock.NewIBCModule(&ibcmock.MockIBCApp{}, scopedIBCMockKeeper)
+
 	app.ICAKeeper = icakeeper.NewKeeper(
 		appCodec, keys[icatypes.StoreKey],
+		app.IBCKeeper.ChannelKeeper, // may be replaced with middleware such as ics29 fee
 		app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
 		app.AccountKeeper, scopedICAKeeper, app.MsgServiceRouter(),
 	)
 	icaModule := ica.NewAppModule(app.ICAKeeper)
-	icaIBCModule := ica.NewIBCModule(app.ICAKeeper, nil)
 
-	// NOTE: the IBC mock keeper and application module is used only for testing core IBC. Do
-	// note replicate if you do not need to test core IBC or light clients.
-	mockModule := ibcmock.NewAppModule(scopedIBCMockKeeper, &app.IBCKeeper.PortKeeper)
-	mockIBCModule := ibcmock.NewIBCModule(&ibcmock.MockIBCApp{}, scopedIBCMockKeeper)
+	// initialize ICA module with mock module as the authentication module on the controller side
+	icaAuthModule := ibcmock.NewIBCModule(&ibcmock.MockIBCApp{}, scopedICAMockKeeper)
+	app.ICAAuthModule = icaAuthModule
+
+	icaIBCModule := ica.NewIBCModule(app.ICAKeeper, icaAuthModule)
```

Comment:
Shouldn't we make the auth module argument optional so we can instantiate an ICA module without enabling the controlling side.

In general, we should make it possible to enable one side of ICA without the other.

Reply:
I don't see a reason to make this optional in our simapp. We should make enabling/disabling one side optional, but I think this will be easier to do after the controller/host split. It should also allow ICA auth modules to be hooked up without being enabled (so gov can enable them dynamically)

Reply:
agreed, im just saying that for a chain that doesn't want to be a controller chain at all. Why should it even pass in an IBC-Auth module just so it can disable it. I'd like to provide a constructor that doesn't require an auth module.

Doesn't need to be used here, i just think that constructor should be provided

Reply:
Great catch @AdityaSripal. I agree that we should be making this optional. There will for sure be chains that don't want to be a controller chain. If it makes sense to do this in a follow-up PR fine with me but if we can make a clear ticket I'd appreciate it :handshake:

Reply:
I'm a little confused. A host chain could use:
```go
    icaIBCModule := ica.NewIBCModule(app.ICAKeeper, nil)
```

Reply:
#523